feat: refactor config, added unittests with mongomock and validation of classes

JackSMunn opened the pull request:
This pull request includes significant changes to the project structure, database models, and testing setup. The most important changes involve moving the database models to a new `db` directory, updating the README to reflect these changes, adding validation logic to the models, and updating the tests to use a mock database.

### Project Structure Changes:
* Moved database models and initialization from `models` to `db` directory. (`.coveragerc`, `MANIFEST.in`, `config.py`, `db/__init__.py`, `db/database.py`, `db/models/__init__.py`, `models/__init__.py`, `models/persist_testing.py`, `models/room.py`, `models/sensor.py`, `pytest.ini`) [[1]](diffhunk://#diff-834e9b406d74791ffbafaeec9cc894082cea9739bc347b6ff9f72312c92ebc79L3-R4) [[2]](diffhunk://#diff-41d5a52589e0480be9c099d2bba7a8135b8b0d71bcbb8df3582a8df1c2295003L3-R3) [[3]](diffhunk://#diff-41d5a52589e0480be9c099d2bba7a8135b8b0d71bcbb8df3582a8df1c2295003R12) [[4]](diffhunk://#diff-117426151e93a626f8b46bfdb3a95b3f4a62e5f4dd6e65975a7c50759bf04482L1-L13) [[5]](diffhunk://#diff-d2f820f6caff6923f172849a1755f80ca386bfafd42e40ef8ba534331be2063dR1-R2) [[6]](diffhunk://#diff-6987bea2e514ac702eb5d26595365f0d42c76840c5336b1ee8dfed541e4bb608R1-R21) [[7]](diffhunk://#diff-61a320bef14adf66c8982b5faea063bba3dec9835770d0a9e91aa5ce4e056797R1-R4) [[8]](diffhunk://#diff-ebdc0efe8893997879ab383f76db51dfb1ed6858333bf793da067a86e08b658fL1-L37) [[9]](diffhunk://#diff-f5713d552fd1062c2e602fb315cbd4c15a9e53b212693795f189cfcfb1cacb1dL1-L75) [[10]](diffhunk://#diff-052f3f91030a40813548bc41902c6930f69b23d14fbf9f16b5ba28b8af7e5368L1-L11) [[11]](diffhunk://#diff-ad8d9d21c821b276300403de4ae3f664c0e640472c9d9f316e5700409ed596f6L1-L13) [[12]](diffhunk://#diff-fb6a686182f16eb54af3c628f38593f347f68aba31de903983023c560288d7a1L2-R3)

### Documentation Updates:
* Updated `README.md` to reflect the new import paths and usage examples for the database models. (`README.md`) [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R7-R15) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L22-R47) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R57-L83)

### Model Enhancements:
* Added validation logic to the `Room` and `Sensor` models to ensure data integrity. (`db/models/room.py`, `db/models/sensor.py`) [[1]](diffhunk://#diff-03118261c76311dc19602e9eca76fffd7675cbcf39511e058fe48d51b787f124R1-R45) [[2]](diffhunk://#diff-857c411fcb05c1ad77f19701c79fd7463cb934ded12f8b7317e24d4fccfeabe1R1-R33)

### Testing Improvements:
* Updated tests to use `mongomock` for mocking MongoDB operations, ensuring tests do not require a live database connection. (`test/crud_test.py`)

### Dependency Updates:
* Added `mongoengine` to `setup.py` dependencies for interacting with MongoDB. (`setup.py`)

MortenMunk committed:
refactor structure

database folder
manifest update
init fixes

JackSMunn committed:
Merge branch 'dev' into config-refactor

MortenMunk committed:
updated init

MortenMunk committed:
fixed module import errors

JackSMunn committed:
test: test for room & sensor validation

- added tests in room_test.py
- added tests in sensor_test.py

JackSMunn committed:
Merge branch 'config-refactor' of https://github.com/SW8-Indoor-Crowded-Regions/Database into config-refactor

MortenMunk committed:
updated manifest, to not include pkg tests

JackSMunn committed:
update: manifest to exclude test

JackSMunn committed:
updated setup.py

- updated setup.py

JackSMunn committed:
fix: setup.py

JackSMunn committed:
docs: updated README.md, reflecting DB usage change

JackSMunn committed:
feat: validation of sensor, room & unittesting crud, validation

- added clean method in both sensor and room classes for validating
- changed sensor to not require movements
- added mongomock to requirements
- added unittests for CRUD operations
- added unittests for room validation
- added unittests for sensor validation.

JackSMunn committed:
fix: linting

JackSMunn committed:
fix: linting

JackSMunn committed:
fix: linting

JackSMunn committed:
fix: linting

JackSMunn committed:
update: added more room types

- changed the different types of rooms that can be used in the enum

Jatewo commented on test/crud_test.py:
Consider making a separate test for each functionality that is being tested.

Jatewo reviewed (approved):
Approved

But consider the separation of tests

JackSMunn committed:
test: refactored crud and sensor tests

- changed sensor class to require rooms
- upped version from 0.1.0 to 0.1.1
- refactored tests in CRUD to be individual tests instead of monolithic test structure (testing many things in each test-case)
- changed testcases in sensor_test to better represent the way sensors are abstracted (2 rooms always)

Mafusn reviewed (approved):
LGMT 🦘
